idaes/core/util/model_diagnostics.py

```diff
@@ -4122,3 +4264,277 @@ def _collect_model_statistics(model):
+            # We want to look for cases where a sum term results in a value much smaller
+            # than the terms of the sum
+            sums = self._sum_combinations(d[0])
+            if any(i <= self._sum_tol * max(d[0]) for i in sums):
+                self.canceling_terms.add(node)
+
+            # Expression is not constant if any child is not constant
+            if not d[1]:
+                const = False
+
+        # Return any problematic terms found
+        return const
+
+    def _check_base_type(self, node):
+        if isinstance(node, VarData):
+            const = node.fixed
+        else:
+            const = True
+        return [value(node)], const, False
+
+    def _check_equality_expression(self, node, child_data):
+        # (In)equality expressions are a special case of sum expressions
+        # First, need to negate one side of the expression
+        mdata = []
+        for i in [0, 1]:
+            if i == 0:
+                vals = []
+                for j in child_data[i][0]:
+                    vals.append(-j)
+                mdata.append((vals, child_data[i][1]))
+            else:
+                mdata.append(child_data[i])
+
+        # Next, call the method to check the sum expression
+        vals, const, _ = self._check_sum_expression(node, mdata)
+
+        # Next, we need to check for canceling terms.
+        # In this case, we can safely ignore expressions of the form constant = sum()
+        # We can also ignore any constraint that is already flagged as mismatched
+        # We will also ignore any constraints where a == b and neither a nor b are sum expressions
+        if not child_data[0][2] and not child_data[1][2]:
+            # Linking constraint, skip checks
+            pass
+        elif node not in self.mismatched_terms and not any(d[1] for d in mdata):
+            # No constant terms, check for cancellation
+            # First, collect terms from both sides
+            t = []
+            for d in mdata:
+                t += d[0]
+
+            # Then check for cancellations
+            sums = self._sum_combinations(t)
+            if any(i <= self._sum_tol * max(t) for i in sums):
+                self.canceling_terms.add(node)
+
+        return vals, const, False
+
+    def _check_general_expr(self, node, child_data):
+        const = self._perform_checks(node, child_data)
+
+        try:
+            # pylint: disable-next=protected-access
+            val = node._apply_operation(
+                list(map(self._get_value_for_sum_subexpression, child_data))
+            )
+        except ValueError:
+            raise ValueError(
+                f"Error in ConstraintTermAnalysisVisitor: error evaluating {str(node)}."
+            )
+        except ZeroDivisionError:
+            raise ZeroDivisionError(
+                f"Error in ConstraintTermAnalysisVisitor: found division with denominator of 0 "
+                f"({str(node)})."
+            )
```

Comment:
Do you want to add a raise in case there is any other error?

Reply:
Is it necessary to have one? I thought that if a different type of exception was encountered it would be raised directly.

Reply:
Necessary? No. Good practice? Yes. It will make the error messages look more standardized / not weirdly disappear into the stack trace.

Reply:
What is the best practices way to do this?

Reply:
```
try:
   (whatever)
except SpecificError:
   (blah)
except Exception as err:
    print(f"Unexpected {err=}, {type(err)=}")
    raise
```
^^ An example!

Reply:
Will pylint complain about the `except Exception`, or does it just complain about `except:`?

Reply:
I think it complains specifically if you `except:` and then don't raise anything. I don't _think_ it yells if you catch general exceptions _and also_ raise them.